wcfsetup/install/files/lib/system/template/plugin/IconFunctionTemplatePlugin.class.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace wcf\system\template\plugin;
+
+use wcf\system\template\TemplateEngine;
+
+/**
+ * Template compiler plugin that embeds icons into the page. The
+ * supported sizes are 16, 24, 32, 48, 64, 96, 128 and 144.
+ *
+ * Usage:
+ *  {icon size=16 name='bell'}
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System\Template\Plugin
+ * @since 6.0
+ */
+final class IconFunctionTemplatePlugin implements IFunctionTemplatePlugin
+{
+    private const SIZES = [16, 24, 32, 48, 64, 96, 128, 144];
+
+    private const TYPES = ['brand', 'solid'];
+
+    /**
+     * @inheritDoc
+     */
+    public function execute($tagArgs, TemplateEngine $tplObj)
+    {
+        $size = \intval($tagArgs['size'] ?? 0);
+        $name = $tagArgs['name'] ?? '';
+        $type = $tagArgs['type'] ?? '';
+
+        if (!\in_array($size, self::SIZES)) {
+            throw new \InvalidArgumentException("An unsupported size `{$size}` was requested.");
+        }
+
+        if ($name === '') {
+            throw new \InvalidArgumentException("The `name` attribute must be present and non-empty.");
+        }
+
+        if ($type !== '' && !\in_array($type, self::TYPES)) {
+            throw new \InvalidArgumentException("An unsupported type `${type}` was specified.");
```

Comment:
```suggestion
            throw new \InvalidArgumentException("An unsupported type `{$type}` was specified.");
```
Your previous syntax is wonky and deprecated with 8.2: https://wiki.php.net/rfc/deprecate_dollar_brace_string_interpolation. Apparently the Syntax checker workflow does not support deprecations as of right now, otherwise it would have caught this. The same is used below in line 50.

Reply:
Good catch, looks like I mixed up the variable interpolation of JavaScript by accident.